swap/common_test.go

```diff
@@ -315,3 +290,45 @@ func (d *dummyMsgRW) ReadMsg() (p2p.Msg, error) {
 func (d *dummyMsgRW) WriteMsg(msg p2p.Msg) error {
 	return nil
 }
+
+// struct used by the testCashoutResultHandler
+type cashChequeDoneData struct {
+	request *CashoutRequest
+	result  *contract.CashChequeResult
+	receipt *types.Receipt
+}
+
+// testCashoutResultHandler is a CashoutResultHandler which writes to a channel after forwarding the result to swap
+type testCashoutResultHandler struct {
+	swap           *Swap
+	cashChequeDone chan cashChequeDoneData
+}
+
+func newTestCashoutResultHandler(swap *Swap) *testCashoutResultHandler {
+	return &testCashoutResultHandler{
+		swap:           swap,
+		cashChequeDone: make(chan cashChequeDoneData),
+	}
+}
+
+// HandleCashoutResult forwards the result to swap if set and afterwards sends it to its channel
+func (h *testCashoutResultHandler) HandleCashoutResult(request *CashoutRequest, result *contract.CashChequeResult, receipt *types.Receipt) error {
+	if h.swap != nil {
+		if err := h.swap.HandleCashoutResult(request, result, receipt); err != nil {
+			return err
+		}
+	}
+	h.cashChequeDone <- cashChequeDoneData{
+		request: request,
+		result:  result,
+		receipt: receipt,
+	}
+	return nil
+}
+
+// helper function to override the cashoutHandler for a cashout processor of swap instance
+func overrideCashoutResultHandler(swap *Swap) *testCashoutResultHandler {
+	cashoutResultHandler := newTestCashoutResultHandler(swap)
+	swap.cashoutProcessor.cashoutResultHandler = cashoutResultHandler
+	return cashoutResultHandler
```

Comment:
Passing it in would make the function itself pointless. Considering this is a test function it should be reasonably clear that this will be overriden by the test handler that is returned.